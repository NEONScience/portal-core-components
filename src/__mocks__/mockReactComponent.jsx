/* eslint react/prop-types: 0 */
/**
   Mock React Component

   Because jest.mock calls can't be dynamic / must hard-code an import path we still need individual
   mocks for every React component that requires mocking. This format defines a generic mock
   implementation for any react component that will appear in snapshots as a div with props mapped
   safely to attributes and any children intact.

   Usage:

   --

   import 'path/to/__mocks__/mockReactComponent';

   jest.mock('path/to/Component', () => mockReactComponent('path/to/Component'));
*/

import React from 'react';

export default function mockReactComponent(path) {
  return (props) => {
    const dataProps = {};
    Object.keys(props).forEach((prop) => {
      if (prop === 'children') { return; }
      const value = props[prop];
      let renderableValue = value;
      try {
        if (typeof value === 'object') { renderableValue = JSON.stringify(value); }
        if (typeof value === 'function') { renderableValue = value.toString(); }
      } catch (error) {
        renderableValue = `Non-renderable ${typeof value} (${error})`;
      }
      dataProps[`prop-${prop}`] = renderableValue;
    });
    const { children } = props;
    return children ? (
      <div ComponentPath={path} {...dataProps}>
        {children}
      </div>
    ) : (
      <div ComponentPath={path} {...dataProps} />
    );
  };
}
