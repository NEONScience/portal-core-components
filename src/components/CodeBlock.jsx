import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import Highlight from 'react-highlight.js';
import '../../node_modules/highlight.js/styles/zenburn.css';

import Theme from '../lib_components/components/Theme/Theme';

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(2, 0),
    '& code': {
      padding: theme.spacing(0, 2),
    },
  },
}));

export default function CodeBlock(props) {
  const classes = useStyles(Theme);
  const { language, children, ...other } = props;
  return (
    <Highlight language={language} className={classes.root} {...other}>
      {children}
    </Highlight>
  );
}

CodeBlock.propTypes = {
  language: PropTypes.string,
  children: PropTypes.string.isRequired,
};

CodeBlock.defaultProps = {
  language: 'typescript',
};
