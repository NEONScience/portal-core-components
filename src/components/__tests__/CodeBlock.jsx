import React from 'react';
import renderer from 'react-test-renderer';

import mockReactComponent from '../../__mocks__/mockReactComponent';

jest.mock('react-highlight.js', () => mockReactComponent('react-highlight.js'));

import CodeBlock from '../CodeBlock';

describe('CodeBlock', () => {
  test('renders with a string', () => {
    const tree = renderer
      .create(
        <CodeBlock>
          var code = 'block';
        </CodeBlock>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with a string and a specified language', () => {
    const tree = renderer
      .create(
        <CodeBlock language="php">
          $code = "block";
        </CodeBlock>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
