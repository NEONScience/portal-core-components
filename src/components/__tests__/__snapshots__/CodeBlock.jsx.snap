// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders with a string 1`] = `
<div
  ComponentPath="react-highlight.js"
  prop-className="makeStyles-root-1"
  prop-language="typescript"
>
  var code = 'block';
</div>
`;

exports[`CodeBlock renders with a string and a specified language 1`] = `
<div
  ComponentPath="react-highlight.js"
  prop-className="makeStyles-root-1"
  prop-language="php"
>
  $code = "block";
</div>
`;
